src/NHibernate/Util/SerializationInfoExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace NHibernate.Util
+{
+	internal static class SerializationInfoExtensions
+	{
+		public static T GetValue<T>(this SerializationInfo info, string name)
+		{
+			if (info == null) throw new ArgumentNullException(nameof(info));
+			return (T) info.GetValue(name, typeof(T));
+		}
+
+		public static void AddValueWithType<T>(this SerializationInfo info, string name, T value)
+		{
+			info.AddValue(name, value, typeof(T));
```

Comment:
`AddValue(string name, object value)` almost does that. Two differences here: 

- The native one will use the actual type of the instance, while the extension method will use the compile time type.
- The native one will default to `object` in case of `null`, which is supported on deserialization (it has a special handling for `null`).

Looking the `AddValueWithType` usages, it does not look to me it needs these differences. In such case I would favor using the native `AddValue(string name, object value)` instead and remove this extension.